securedrop/journalist_app/api.py

```diff
@@ -0,0 +1,221 @@
+from datetime import datetime
+from functools import wraps
+import json
+
+from flask import abort, Blueprint, current_app, jsonify, request, send_file
+
+from db import db
+from journalist_app import utils
+from models import Journalist, Reply, Source, Submission
+from store import NotEncrypted
+
+
+def token_required(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        try:
+            auth_header = request.headers['Authorization']
+        except KeyError:
+            return abort(403, 'API token not found in Authorization header.')
+
+        if auth_header:
+            auth_token = auth_header.split(" ")[1]
+        else:
+            auth_token = ''
+        if not Journalist.verify_api_token(auth_token):
+            return abort(403, 'API token is invalid or expired.')
+        return f(*args, **kwargs)
+    return decorated_function
+
+
+def get_or_404(model, object_id, column=''):
+    if column:
+        result = model.query.filter(column == object_id).first()
+    else:
+        result = model.query.get(object_id)
+    if result is None:
+        abort(404)
+    return result
+
+
+def make_blueprint(config):
+    api = Blueprint('api', __name__)
+
+    @api.route('/')
+    def get_endpoints():
+        endpoints = {'sources_url': '/api/v1/sources/',
+                     'current_user_url': '/api/v1/user/',
+                     'submissions_url': '/api/v1/submissions/',
+                     'auth_token_url': '/api/v1/token/'}
+        return jsonify(endpoints), 200
+
+    @api.route('/token/', methods=['POST'])
```

Comment:
This should not have a trailing slash. The docs also specify that is does not have a trailing slash.